plugin-niassa/src/main/java/ca/on/oicr/gsi/shesmu/niassa/NiassaServer.java

```diff
@@ -165,6 +165,9 @@ public DirectoryAndIniCache(Path fileName) {
       final MetadataWS metadata = metadataConstructor.get();
       final WorkflowRun run = metadata.getWorkflowRun(key.intValue());
       metadata.clean_up();
+      if (run == null) {
+        return null;
```

Comment:
Looking further down the method, there are multiple `return Optional.empty()` , even in the case of catching an exception (line 202). Returning `Optional.empty()` here might be more idiomatic, and calling code already knows to check for empty optional.